tests/test_other.py

```diff
@@ -7664,3 +7664,59 @@ def test_native_link_error_message(self):
     Popen([LLVM_AR, 'r', 'hello_world.a', 'hello_world.o'], env=get_clang_native_env(), stdout=PIPE, stderr=PIPE).communicate()
     out, err = Popen([PYTHON, EMCC, 'hello_world.a', '-o', 'hello_world.js'], stdout=PIPE, stderr=PIPE).communicate()
     assert 'exists but was not an LLVM bitcode file suitable for Emscripten. Perhaps accidentally mixing native built object files with Emscripten?' in err
+
+  def test_single_file(self):
+    with clean_write_access_to_canonical_temp_dir():
+      for single_file_enabled in [True, False]:
+        for meminit1_enabled in [True, False]:
+          for debug_enabled in [True, False]:
+            for emterpreter_enabled in [False]:
+              for wasm_enabled in [True, False]:
+                for asmjs_fallback_enabled in [True, False]:
+                  # skip unhelpful option combinations
+                  if (
+                    (asmjs_fallback_enabled and not wasm_enabled)
+                  ):
+                    continue
+
+                  expect_asmjs_code = asmjs_fallback_enabled and wasm_enabled
+                  expect_emterpretify_file = emterpreter_enabled
+                  expect_meminit = (meminit1_enabled and not wasm_enabled) or (wasm_enabled and asmjs_fallback_enabled)
+                  expect_wasm = wasm_enabled
+                  expect_wast = debug_enabled and wasm_enabled
+
+                  cmd = [PYTHON, EMCC, path_from_root('tests', 'hello_world.c')]
+
+                  if single_file_enabled:
+                    expect_asmjs_code = False
+                    expect_emterpretify_file = False
+                    expect_meminit = False
+                    expect_wasm = False
+                    expect_wast = False
+                    cmd += ['-s', 'SINGLE_FILE=1']
+                  if meminit1_enabled:
+                    cmd += ['--memory-init-file', '1']
+                  if debug_enabled:
+                    cmd += ['-g4']
+                  if emterpreter_enabled:
+                    cmd += ['-s', 'EMTERPRETIFY=1', '-s', "EMTERPRETIFY_FILE='a.out.dat'"]
+                  if wasm_enabled:
+                    cmd += ['-s', 'WASM=1']
+                  if asmjs_fallback_enabled:
+                    cmd += ['-s', "BINARYEN_METHOD='native-wasm,asmjs'"]
+
+                  for x in os.listdir('.'):
+                    if x.startswith('emcc-') or x.startswith('a.out'):
+                      os.unlink(x)
```

Comment:
Is this supposed to be needed? Originally I'd had `with clean_write_access_to_canonical_temp_dir():` inside the loop, expecting all the files to be wiped between iterations, but that turned out not to be the case.